src/raft/raft.go

```diff
@@ -243,20 +266,22 @@ func (rf *Raft) AppendEntries(args *AppendEntriesArgs, reply *AppendEntriesReply
 // not the struct itself.
 //
 // sendRequestVote sends a RequestVote RPC to a server.
-// serverIdx is the index of the target server in rf.peers[].
 func (rf *Raft) sendRequestVote(wg *sync.WaitGroup, serverIdx int, nVotes *int, nConnected *int) {
 	defer wg.Done()
+	lastLog := rf.getLastLogEntry()
+
 	reqVoteArgs := RequestVoteArgs{
-		Term:        rf.currentTerm,
-		CandidateId: rf.myIdx,
-		//LastLogIndex: 0,
-		//LastLogTerm: 0,
+		Term:         rf.currentTerm,
+		CandidateId:  rf.myIdx,
+		LastLogIndex: lastLog.Idx,
+		LastLogTerm:  lastLog.Term,
 	}
 	reqVoteReply := RequestVoteReply{}
 
 	ok := false
 	done := make(chan bool)
 	go func() {
+		// serverIdx is the index of the target server in rf.peers[]
 		ok = rf.peers[serverIdx].Call("Raft.RequestVote", &reqVoteArgs, &reqVoteReply)
```

Comment:
The part of logic from the line 281 to 287 could be encapsulated into one function:
```suggestion
func (rf *Raft) call(rpcName string, args interface{}, reply interface{}, ok *bool, done chan bool) {
    ......
}
```